### PR TITLE
Be able to type diagram and events

### PR DESCRIPTION
**Pull request by nikku**
This allows implementors to type the diagram (available services) and eventBus events (available events):

### Sublime Text

capture hj3ree_optimized

### VSCode

capture WrEJZW_optimized

### CODE REVIEW

File: lib/core/EventBus.js [hunk 1]
```diff
@@ -131,8 +133,9 @@ export default function EventBus() {
   this.on('diagram.destroy', 1, this._destroy, this);
 }
 
-
 /**
+ * @overlord
```
Comment (vsgoulart) [inline lib/core/EventBus.js:137]: @nikku  I'm not familiar with this keyword and couldn't find it anywhere? isn't the correct one `@overload`?

Comment (nikku) [reply lib/core/EventBus.js:137]: This keyword is the working version of TypeScript `@overload`, cf. [documentation](https://github.com/nikku/bio-dts?tab=readme-ov-file#features).

Comment (nikku) [reply lib/core/EventBus.js:137]: If you want to use `@overlord`, then you'd need to use `bio-dts` to generate the types. 

We do it in core bpmn.io as our sources remain `.js` and we still want to have clean types.

**Comment by barmac** [comment]
I will have a look at this today.

**Review by barmac — APPROVED** [review]
This works great. I can't wait to see that in action in bpmn-js.

**Pull request merged**